tutorials/week4-react.md

```diff
@@ -0,0 +1,689 @@
+layout: page
+title: React Tutorial
+permalink: /tutorials/week4-react-basics
+parent: Tutorials
+nav_order: 6
+---
+
+# React Basics
+This tutorial covers the basic concepts of react. By the end of this tutorial, you will be able to create a new react app, understand the basic concepts of react such as states and props, understand React hooks and handling events.
+
+-   [React Basics](#react-basics)
+-   [Creating a new NextJs App](#creating-a-new-next-js-app)
+-   [Understanding a React App](#react-component)
+    -   [Components](#react-component)
+    -   [Template For Structure of Function Component](#template-for-structure-of-function-component)
+    -   [Props](#props)
+    -   [State](#state)
+    -   [Communicating between Components](#communication-between-components)
+        - [Parent to Child](#parent-to-child-component-communication)
+        - [Child to Parent](#child-to-parent-component-communication)
+    -   [Handling Events](#handling-events)
+-   [React Hooks](#react-hooks)
+    -   [UseState](#usestate)
+        - [Updating Arrays in State](#updating-arrays-in-state)
+    -   [UseEffect](#useeffect)
+        - [Object Dependencies in UseEffect](#object-dependencies-in-useEffect)
+
+
+## Creating a New Next Js App
+
+Let's use **npx** and **create-next-app** to create a new Next.Js project 
+
+-   **npx** stands for Node Package Execute (Part of the npm package since version 5.2). It is a runner that can execute any package that you want from the npm registry without even installing that package. In this case npx tool temporarily installs the create-next-app npm package and uses it to create our project.
+
+
+- Please check the following reference to the Next.Js boiler plate and getting started with framework 
+[Next.Js official documentation](https://nextjs.org/learn/react-foundations)
+
+1. We use the create-next-app npx package to create an application that creates a fully-featured TS package.
+
+    ```bash
+    npx create-next-app my-app
+    ```
+
+    - Note 1: This will create a new project directory called my-app under the current directory.
+    - Note 2: This will create a git repo in my-app, so you probably shouldn't do this inside a pre-existing git repository.
+    - Note 3: Create the project with the following details after executing the command
+    ![image](./assets/week4-react/start-details.png)
+
+2. Navigate to the project directory using the command:
+    ```bash
+    cd my-app
+    ```
+3. To start the development server for React, run the command:
+    ```bash
+    npm run dev
+    ```
+    - Note: Next.Js renders the content within the `app/pages.tsx` file as the root component.
+
+4. Navigate to http://localhost:3000/ to see the default react page.
+    - After a few seconds, a browser window opens, with our app running:
+      ![image](./assets/week4-react/start.png)
+5. [Chakra UI components](https://chakra-ui.com) can be reused and the library can be installed for Next.js with following command:
```

Comment:
I think Chakra definition should be changed to:

5. [Chakra UI](https://chakra-ui.com) provides a comprehensive library of reusable React components for developing modern web applications. To install Chakra UI for Next.js, please enter the following command: